axum/src/routing/mod.rs

```diff
@@ -300,6 +303,14 @@ where
             path = "/";
         }
 
+        if path == "/" {
+            if self.service_nested_at_root {
+                panic!("Cannot nest more than one service at `/`. Consider using `Router::merge` instead.");
```

Comment:
Maybe add "if applicable"? Since this is only possible if the nested service is another router. Alternatively, we could have this check in `nest` instead of `nest_service`.